dask_planner/src/sql.rs

```diff
@@ -152,6 +152,31 @@ impl ContextProvider for DaskSQLContext {
                 let rtf: ReturnTypeFunction = Arc::new(|_| Ok(Arc::new(DataType::Int64)));
                 return Some(Arc::new(ScalarUDF::new(name, &sig, &rtf, &fun)));
             }
+            "last_day" => {
+                let sig = Signature::exact(
+                    vec![DataType::Timestamp(TimeUnit::Nanosecond, None)],
+                    Volatility::Immutable,
+                );
+                let rtf: ReturnTypeFunction =
+                    Arc::new(|_| Ok(Arc::new(DataType::Timestamp(TimeUnit::Nanosecond, None))));
+                return Some(Arc::new(ScalarUDF::new(name, &sig, &rtf, &fun)));
+            }
+            "timestampadd" => {
+                let sig = Signature::exact(
+                    vec![DataType::Utf8, DataType::Int64, DataType::Date64],
```

Comment:
In the tests, I am getting the following optimizer warnings:

```
Skipping optimizer rule type_coercion due to unexpected error: Error during planning: Coercion from [Utf8, Int64, Timestamp(Nanosecond, None)] to the signature Exact([Utf8, Int64, Date64]) failed.
```

I would've assumed that we would be able to coerce a `Timestamp` into a `Date64`, but in general think that since they both have roughly the same corresponding Python type that we should probably allow for both signatures.

@andygrove is there any way we can easily define signatures that parametrize on the type of one or more inputs? In this case, we would probably want this signature to be `OneOf` with several signatures for the various int / date types we can accept